actionpack/test/controller/action_pack_assertions_test.rb

```diff
@@ -170,6 +176,16 @@ def test_string_constraint
     end
   end
 
+  def test_with_routing_works_with_api_only_controllers
+    @controller = ApiOnlyController.new
```

Comment:
Alternatively, I think we could skip the `ApiOnlyController` class with: `@controller = Class.new(ActionController::Api).new`